### PR TITLE
T142: Create database seed with a test user and friends

### PR DESCRIPTION
**Pull request by kdata12**
## Description 

Created a database seed that will insert one test user and friend relationships with other dummy users. 

## Types of changes

What types of changes does your code introduce to WeWrap?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [x] I have made sure that eslint was passed locally
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: backend/prisma/userSeed.ts
Comment (kdata12) [inline backend/prisma/userSeed.ts]: So I discovered that the `crypto.pbkdf2` fn takes some time to create a hash value, and afterward, create a test user record. This setTimeout solution guarantees that a test user record was created before moving on to the next step.

File: backend/prisma/userSeed.ts
Comment (justinpchen94) [inline backend/prisma/userSeed.ts]: i won't really block this pr since this is just a db seed script but consider how you could've just written the original production fn in a way that is exportable in a fn such as `createUserWithHashedPw` and then just called that here instead, rather than copy pasting the logic

File: backend/prisma/userSeed.ts
Comment (justinpchen94) [reply backend/prisma/userSeed.ts]: while this is probably fine given this is a seed script, in reality you are mixing Callback style js programming with async/await. due to the inherently async nature of js, your line 64 begins an asynchronous execution of code but does not block the rest of this script from running. the best thing you could do here is use a promisification library such as bluebird to asyncify the crypto.pbkdf2 call and convert the callback into a promise. alternatively I think you could actually do it without al library but this presents an opportunity to look into those libraries if you wish to

File: backend/prisma/userSeed.ts
Comment (justinpchen94) [inline backend/prisma/userSeed.ts]: part of the problem here too is that the .map call is not asynchronous, so it is not going to pause execution of js on this entire .map call finishing. it will queue up all of these async fn's and they will just kind of finish whenever they happen to finish. you could try looking into using Promise.all here

File: backend/prisma/userSeed.ts
Comment (justinpchen94) [inline backend/prisma/userSeed.ts]: typo

File: backend/prisma/userSeed.ts
Comment (kdata12) [reply backend/prisma/userSeed.ts]: i will keep this in mind next time I copy and paste code

File: backend/prisma/userSeed.ts
Comment (kdata12) [reply backend/prisma/userSeed.ts]: Got this taken care of by promisifying the cypto function

File: backend/prisma/userSeed.ts
Comment (kdata12) [reply backend/prisma/userSeed.ts]: Fixed this with promise.all()

File: backend/prisma/userSeed.ts
Comment (justinpchen94) [reply backend/prisma/userSeed.ts]: i love it, good shit!

File: backend/prisma/userSeed.ts
Comment (justinpchen94) [inline backend/prisma/userSeed.ts]: nit: it's not unknown, it should be a `User` right?

**Review by justinpchen94 — APPROVED** [review]
lgtm pending 1 comment

File: backend/prisma/userSeed.ts
Comment (kdata12) [reply backend/prisma/userSeed.ts]: fixed! with `promise<User>`

**Pull request merged**
